### ISSUE TITLE
Parse function breaks when there’s a line ending in a string 

### ISSUE THREAD

**Issue by ariasuni** [rspivak/slimit#100]
```
In: tree = parser.parse('var i = "test\nvalue"')
Illegal character '"' at 1:8 after LexToken(EQ,'=',1,6)
Illegal character '"' at 1:19 after LexToken(ID,'value',1,14)

In: tree.to_ecma()
'var i = test;\nvalue;'
```

The behavior is the same with `\r`.

**Comment by metatoaster** [comment]
This is fixed in [`calmjs.parse`](https://pypi.org/project/calmjs.parse).

```
>>> from calmjs.parse import es5
>>> es5('var i = test;\nvalue;')
<ES5Program @1:1 ?children=[
  <VarStatement @1:1 ?children=[
    <VarDecl @1:5 identifier=<Identifier ...>, initializer=<Identifier ...>>
  ]>,
  <ExprStatement @2:1 expr=<Identifier @2:1 value='value'>>
]>
```

**Comment by metatoaster** [comment]
No, never mind, if it's a string token and actual new line, nodejs doesn't even like it either.

```
$ cat | node
var i = "test
value"
[stdin]:1
var i = "test
        ^^^^^

SyntaxError: Invalid or unexpected token
```

ES5 (which is what slimit supports) doesn't have multiline strings like Python does.

**Comment by ariasuni** [comment]
Well I had this problem when trying to scrape information out of a working JavaScript code on a high-traffic website.

**Comment by metatoaster** [comment]
Can you please provide the link to the example that choked?

**Comment by metatoaster** [comment]
Anyway, I do see what you mean - I had mistakenly used my patched version of slimit that correctly reported that as a parsing error.  Anyway, the correct behavior with that input should throw a SyntaxError exception, which my patched version (and `calmjs.parse`) does. The definition in the ECMA-262 specification that states this as an invalid syntax is defined in [section 7.8.4](https://www.ecma-international.org/ecma-262/5.1/#sec-7.8.4) (specifically "A line terminator character cannot appear in a string literal" at the bottom of that section, where a "line terminator" includes newline characters)

To make things most clear, this is the input JavaScript with the invalid syntax:

```javascript
var i = "test
value"
```
Assume that input is assigned to `program` in the following Python code:

```python
>>> from slimit.parser import Parser
>>> parser = Parser()
>>> node = parser.parse(program)
Illegal character '"' at 1:8 after LexToken(EQ,'=',1,6)
Illegal character '"' at 1:19 after LexToken(ID,'value',1,14)
>>> print(node.to_ecma())
var i = test;
value;
```

This changed the program entirely, and this is where my initial confusion lied.  The correct behavior is implemented in `calmjs.parse`, which correctly process this as a syntax error:

```python
>>> from calmjs.parse import es5
>>> es5(program)
Traceback (most recent call last):
...
calmjs.parse.exceptions.ECMASyntaxError: Illegal character '"' at 1:9 after '=' at 1:7
```

**Comment by ariasuni** [comment]
Well, I’m probably mistaken: this should have been a non-working JavaScript extract among the working ones, because I have the same kind of error with newlines inside strings in my web browser’s console.
